app/controllers/comments_controller.rb

```diff
@@ -1,20 +1,23 @@
 class CommentsController < ApplicationController
-    def create
-      @article = Article.find(params[:article_id])
-      @comment = @article.comments.create(comment_params)
-      redirect_to article_path(@article)
-    end
 
-    def destroy
-      @article = Article.find(params[:article_id])
-      @comment = @article.comments.find(params[:id])
-      @comment.destroy
-      redirect_to article_path(@article), status: :see_other
-    end
+  http_basic_authenticate_with name: "dhh", password: "secret", only: :destroy
```

Comment:
What does http_basic_authenticate_with do? is it obtained from any gem?

Reply:
`http_basic_authenticate_with_do` method allows access to the requested action if that method allows it and acts as a before filter which accepts a name and password